### DISCUSSION TITLE
How to integrate fzf?

### DISCUSSION THREAD

**Discussion by fikovnik** [zimfw/zimfw#537] [Q&A] [answered]
I'm pretty sure I missing something, but I'm not sure how to properly integrate fzf.
I installed it using homebrew and added `zmodule fzf` into `.zimrc`.
Do I still need to add manually the `source /opt/homebrew/Cellar/fzf/0.53.0/shell/*.zsh`?
Without it I do not get any of the keybindings / compleation working.

**Comment by antoineco** [comment]
There is no need to source anything manually, the module is sourced automatically.

The main purpose of Zim's fzf module is to set sensible defaults for fzf to call out to external programs like `rg`, `bat`, etc. in case you have any of these installed.

For instance, I have both `rg` and `bat` installed on my machine, so Zim exports the following variables for me:

```console
$ echo $FZF_DEFAULT_COMMAND
command rg -uu -g '!.git' --files --no-messages
```

```console
$ echo $FZF_CTRL_T_OPTS
--bind ctrl-/:toggle-preview --preview 'command bat --color=always --line-range :500 {}'
```

Could you please confirm that:
1. You have any of the [relevant tools](https://github.com/zimfw/fzf) installed in your PATH
1. `printenv | grep FZF_` does not print anything

**Comment by fikovnik** [comment]
Thank you, I have `fd`, `rg` and `bat` installed.

Here is the output
```
printenv | grep FZF_
FZF_ALT_C_COMMAND=command fd -H --no-ignore-vcs -E .git -td
FZF_ALT_C_OPTS=--bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}' --bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}' --bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}' --bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}' --bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}' --bind ctrl-/:toggle-preview --preview 'command ls --color=always -CF {}'
FZF_CTRL_T_COMMAND=command fd -H --no-ignore-vcs -E .git -td -tf
FZF_DEFAULT_COMMAND=command fd -H --no-ignore-vcs -E .git -td -tf
```

But still no keybinding is set.

My `.zimrc` is:

```
zmodule environment
zmodule git
zmodule input
zmodule termtitle
zmodule utility
zmodule fzf
zmodule archive
zmodule homebrew
zmodule joke/zim-chezmoi
zmodule romkatv/powerlevel10k
zmodule zsh-users/zsh-completions --fpath src
zmodule completion
zmodule zsh-users/zsh-syntax-highlighting
zmodule zsh-users/zsh-history-substring-search
zmodule zsh-users/zsh-autosuggestions
```

**Reply by ericbn** [reply]
Looks like there's an issue with the value of FZF_ALT_C_OPTS. It has `-bind ctrl-\/:toggle-preview --preview 'command ls --color=always -CF {}'` repeated 6 times, and it should be there only once.

**Comment by fikovnik** [comment, marked as answer]
Thanks a lot. In the end, the problem was that fzf was before homebrew so it did not find the executable.

**Reply by aronnoordhoek** [reply]
Nice, this did it for me too. Love it when people post their own solutions.
